source/core/module_coreHelpers.js

```diff
@@ -303,37 +303,40 @@
             return currentFPS;
         };
 
-        var writeNewError = function (newErrorStatus, newErrorCode, newErrorSource, existingErrorStatusPointer, existingErrorCodePointer, exisitingErrorSourcePointer) {
-            Module.eggShell.dataWriteBoolean(existingErrorStatusPointer, newErrorStatus);
-            Module.eggShell.dataWriteInt32(existingErrorCodePointer, newErrorCode);
-            Module.eggShell.dataWriteString(exisitingErrorSourcePointer, newErrorSource);
+        var writeNewError = function (errorValueRef, newError) {
+            Module.eggShell.writeJSON(errorValueRef, JSON.stringify(newError));
```

Comment:
Since we know the structure of an error can we do use the more optimized eggShell functions instead?
findSubValueRef for the different components (status, code, source) and write to them using the correct eggShell function (writeDouble, writeDouble, writeString)

Reply:
+1